File: snail-race-bet/java/src/main/java/snail/race/kata/adapters/BetRepositoryMongoDb.java
package snail.race.kata.adapters;

import com.mongodb.client.MongoDatabase;
import snail.race.kata.domain.Bet;
import snail.race.kata.domain.BetRepository;

import java.util.ArrayList;
import java.util.List;

public class BetRepositoryMongoDb implements BetRepository {
    private final MongoDatabase database;

    public BetRepositoryMongoDb(MongoDatabase database) {
        this.database = database;
    }

    @Override
    public void register(Bet bet) {
        database.getCollection("bet", Bet.class).insertOne(bet);
    }

    @Override
    public List<Bet> findByDateRange(long from, long to) {
        return database.getCollection("bet", Bet.class).find().into(new ArrayList<>());
    }
}

File: snail-race-bet/typescript/src/adapters/BetRepositoryMongoDb.ts
import {Db} from "mongodb";
import {Bet, BetRepository, PodiumPronostic} from "../domain/BetRepository";

export class BetRepositoryMongoDb implements BetRepository {
    constructor(private db: Db) {
    }

    async register(bet: Bet): Promise<void> {
        await this.getCollection().insertOne({
            ...bet
        })
    }

    async findByDateRange(from: number, to: number): Promise<Bet[]> {
        return await this.getCollection()
            .find({
                timestamp: {$gte : from, $lt:to}
            })
            .map(doc => new Bet(
                doc.gambler,
                new PodiumPronostic(doc.pronostic.first, doc.pronostic.second, doc.pronostic.third),
                doc.timestamp
            ))
            .toArray()
    }

    private getCollection() {
        return this.db.collection('bets');
    }
}